compiler/crates/graphql-ir/src/errors.rs

```diff
@@ -539,6 +536,13 @@ pub enum ValidationMessage {
     ResolverInMutation,
 }
 
+#[derive(Clone, Debug)]
+pub struct ValidationDiagnosticCode;
+
+impl ValidationDiagnosticCode {
+    pub const EXPECTED_OPERATION_VARIABLE_TO_BE_DEFINED: i32 = 1;
```

Comment:
I'm not a fan of this being just an integer. Since the diagnostic code is also shown in the LSP client, ideally it would be something like `relay001`. But having a string here will likely mean that we can no longer pattern match. Maybe there's something like a macro we can utilize on the errors enum to define a diagnostic code there and we can easily reference it and it gets properly formatted when sent to the client?